### ISSUE TITLE
DahuaSunriseSunset.exe stops responding after pressing "Simulate xy"

### ISSUE THREAD

**Issue by Lecso11** [bp2008/DahuaSunriseSunset#14]
I run the service on a virtual machine, I could set up my config, but when I want to test it, and press Simulate Sunrise or Sunset, the .exe stops respondig, and nothing happens.
![image](https://github.com/user-attachments/assets/35e64642-3e5e-436e-bbf8-bcfa6f79ccf8)

**Comment by bp2008** [comment]
Hmm.  Given the 0% CPU usage of the program while it is hanging, I suspect your configuration is pointing at a camera address that is not responding.  I probably did not go to the effort of making the simulate buttons operate on a background thread, so while the web requests are active/retrying, the user interface would freeze.    It should eventually unfreeze but could take a few minutes.  An error message should appear in the log file in the folder where you ran the exe.

**Comment by Lecso11** [comment]
> Hmm. Given the 0% CPU usage of the program while it is hanging, I suspect your configuration is pointing at a camera address that is not responding. I probably did not go to the effort of making the simulate buttons operate on a background thread, so while the web requests are active/retrying, the user interface would freeze. It should eventually unfreeze but could take a few minutes. An error message should appear in the log file in the folder where you ran the exe.

Hi! You were right, it could not connect to the camera. If I don't put a port after the IP or I put 80, I got the error 401 unauthorized. If I use the https port, I got the "could not connect" error. What am I doing wrong?
<img width="960" alt="image" src="https://github.com/user-attachments/assets/169dc79a-e372-402b-9074-94360344662d">

**Comment by bp2008** [comment]
`401 unauthorized` means at least the program was able to get a response from the camera, so the address must be fine.  But the user name or password may be wrong or the camera may just not support the necessary API.

**Comment by Lecso11** [comment]
> `401 unauthorized` means at least the program was able to get a response from the camera, so the address must be fine. But the user name or password may be wrong or the camera may just not support the necessary API.

Credentials are right, I checked it multiple times. If the API is not supported, there is nothing I can do?

**Comment by bp2008** [comment]
There would be nothing you can do via this app, short of modifying the source code to work with whatever your camera supports.

**Comment by Lecso11** [comment]
Sadly I don't have the skillset to do that, can you help me maybe?

**Comment by bp2008** [comment]
Nope.  I do not work on this project anymore as I haven't used it in years, and I have no way of testing API requests against your camera to be able to figure out what works and what doesn't.
